### ISSUE TITLE
The CustomResourceDefinition "solrclouds.solr.apache.org" is invalid

### ISSUE THREAD

**Issue by arogic** [apache/solr-operator#502]
Hi,

I am trying to deploy the Manifest for the CRD's called "all-with-dependencies.yaml" (https://solr.apache.org/operator/downloads/crds/v0.6.0/all-with-dependencies.yaml) for version "0.6.0", but getting the below error:

```
$ kubectl apply -f all-with-dependencies.yaml
customresourcedefinition.apiextensions.k8s.io/solrbackups.solr.apache.org configured
customresourcedefinition.apiextensions.k8s.io/solrprometheusexporters.solr.apache.org configured
customresourcedefinition.apiextensions.k8s.io/zookeeperclusters.zookeeper.pravega.io configured
The CustomResourceDefinition "solrclouds.solr.apache.org" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

If I try to use kubectl replace I get the following error:
```
$ kubectl replace -f all-with-dependencies.yaml
customresourcedefinition.apiextensions.k8s.io/solrbackups.solr.apache.org replaced
customresourcedefinition.apiextensions.k8s.io/solrprometheusexporters.solr.apache.org replaced
customresourcedefinition.apiextensions.k8s.io/zookeeperclusters.zookeeper.pravega.io replaced
Error from server (NotFound): error when replacing "all-with-dependencies.yaml": customresourcedefinitions.apiextensions.k8s.io "solrclouds.solr.apache.org" not found
```

Is there a way to get around this issue without splitting the manifest into smaller pieces?

Thanks in advance

**Comment by HoustonPutman** [comment]
Yes, from the begging I would recommend using the command here: https://artifacthub.io/packages/helm/apache-solr/solr-operator#installing-the-chart

```
kubectl create -f https://solr.apache.org/operator/downloads/crds/v0.6.0/all-with-dependencies.yaml
```

and for upgrades, do the `replace` command.

Since you already have 3/4 of the crds installed, create will give you warnings about the first 3, but it will install the the fourth just fine.

**Comment by anilkhichar** [comment]
@arogic @HoustonPutman The metadata size error can be removed by taking out all the description content from the CRD using [yq](https://mikefarah.gitbook.io/yq/).

```
solr_operator_version=0.6.0
wget -q https://solr.apache.org/operator/downloads/crds/v$solr_operator_version/all.yaml -O - \
  | yq eval-all 'del(.. | select(has("description")).description)' \
  | kubectl apply -f -

customresourcedefinition.apiextensions.k8s.io/solrbackups.solr.apache.org configured
customresourcedefinition.apiextensions.k8s.io/solrclouds.solr.apache.org configured
customresourcedefinition.apiextensions.k8s.io/solrprometheusexporters.solr.apache.org configured
```

**Comment by anilkhichar** [comment]
### kubectl create/replace VS kubectl apply/apply
Can we get real data points specific to solr operator CRDs where using apply can get wrong. Based on my experiment there is no difference in the applied CRD content.

My experiment steps:
Step1:
1. Install v0.5.1 CRD using kubeclt create -f
`kubeclt get crd -o yaml > *.yaml`
1. Upgrade to v0.6.0 CRD using kubeclt replace -f
`kubeclt get crd -o yaml > *.yaml`
1. Upgrade to v0.7.0 CRD using kubeclt replace -f
`kubeclt get crd -o yaml > *.yaml`

Step2:
1. Install v0.5.1 CRD using kubeclt apply -f
`kubeclt get crd -o yaml > *.yaml`
1. Upgrade to v0.6.0 CRD using kubeclt apply -f
`kubeclt get crd -o yaml > *.yaml`
1. Upgrade to v0.7.0 CRD using kubeclt apply -f
`kubeclt get crd -o yaml > *.yaml`

Step3, Step4: Also repeated steps from v0.3.0 to v0.7.0 (but same result)

Now the generated outcome YAML files do have exact same content except one `last-applied-cofiguration` extra element in the apply. And we don't need to worry about it.

![kubectl_apply_vs_replace_experiment](https://github.com/apache/solr-operator/assets/5746468/e06d8824-1ac9-4f8c-9b1f-ce5601d3422f)
![repalce_vs_apply_diff](https://github.com/apache/solr-operator/assets/5746468/20f63d74-5d73-4475-adad-7e6ce81a4375)
